ts/src/kraken.ts

```diff
@@ -648,17 +472,82 @@ export default class kraken extends Exchange {
         return result;
     }
 
+    async fetchWithdrawalMethods () {
+        let withdrawalNetworks = this.safeValue (this.options, 'withdrawalNetworks', []);
+        if (withdrawalNetworks.length === 0 && this.checkRequiredCredentials (false)) {
+            try {
+                const withdrawNetworkResponse = await this.privatePostWithdrawMethods ();
+                // Withdrawal Methods
+                // {
+                //     "error": [],
+                //     "result": [
+                //       {
+                //         "asset": "XXBT",
+                //         "method": "Bitcoin",
+                //         "network": "Bitcoin",
+                //         "minimum": "0.0004"
+                //       },
+                //       {
+                //         "asset": "XXBT",
+                //         "method": "Bitcoin Lightning",
+                //         "network": "Lightning",
+                //         "minimum": "0.00001"
+                //       }
+                //     ]
+                //  }
+                this.handleErrors (null, null, null, null, null, {}, withdrawNetworkResponse, null, null);
+                withdrawalNetworks = this.safeValue (withdrawNetworkResponse, 'result', []);
+                this.options['withdrawalNetworks'] = withdrawalNetworks;
+            } catch (e) {
+                // The user does not have access to the endpoint, not a big deal we just return an empty array
+                if (e instanceof PermissionDenied) {
+                    return [];
+                }
+                throw e;
+            }
+        }
+        this.options['methodsByEcids'] = {};
+        this.options['ecidsByMethods'] = {};
+        const networksPerCurrency = {};
+        for (let i = 0; i < withdrawalNetworks.length; i++) {
+            const network = this.safeValue (withdrawalNetworks, i);
+            const asset = this.safeString (network, 'asset');
+            const code = this.safeCurrencyCode (asset);
+            const networkInformation = {
+                'info': network,
+                'network': this.safeString (network, 'network'),
+                'withdraw': {
+                    'min': this.parseNumber (this.safeString (network, 'minimum')),
+                    'max': undefined,
+                },
+                'withdraw_enabled': true,
```

Comment:
Are you sure?

Reply:
since the endpoint is called `withdrawal_methods` and it does not return a flag regarding networks status, there is no other way to get this information and I assume it's true

Reply:
when I say I, I mean "we (should) assume"